### ISSUE TITLE
Starting with non-semver-tag, Keel is upgrading to any tag

### ISSUE THREAD

**Issue by v-morlock** [keel-hq/keel#480]
When we first setup an environment, we start by deploying a "latest" image. Starting from that, Keel is upgrading to any tag, not only to semver-Tags, e.g. from "latest" to "dev".

We specified the following policy:

```
      "keel.sh/policy"    = "major"
      "keel.sh/trigger"   = "pubsub"
```

Am I missing anything here?

**Comment by rusenask** [comment]
that is strange! it should update to the latest semver from `latest`, not just any tag. I guess you are using it with polling trigger?

**Comment by v-morlock** [comment]
Nope, I'm using the gcr pubsub trigger. Lately I also experienced it to show the wrong version as a target, e.g. an image was pushed as _image:1.0.1_ but keel tried (and failed) to evaluate on just _image_

Thanks in advance!

**Comment by rusenask** [comment]
would it be possible for you to get a sample of this gcr message?
